packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -4524,96 +4521,15 @@ describe('StripeHelper', () => {
-      sandbox
-        .stub(stripeHelper, 'extractCustomerDefaultPaymentDetails')
-        .resolves({ country: 'GD' });
-      sandbox.stub(stripeHelper, 'updateCustomerBillingAddress').resolves();
-      const actual = await stripeHelper.setCustomerLocation(customer1);
-      assert.isFalse(actual);
-      sinon.assert.calledOnceWithExactly(
-        stripeHelper.extractCustomerDefaultPaymentDetails,
-        customer1
-      );
-      sinon.assert.calledOnce(Sentry.withScope);
-      sinon.assert.calledOnceWithExactly(
-        sentryScope.setContext,
-        'setCustomerLocation',
-        { customer: { id: customer1.id } }
-      );
-      sinon.assert.calledOnceWithExactly(
-        Sentry.captureMessage,
-
-        `Cannot find a postal code or country for customer ${customer1.id}`,
-        Sentry.Severity.Error
-      );
-      sinon.assert.notCalled(stripeHelper.googleMapsService.getStateFromZip);
-      sinon.assert.notCalled(stripeHelper.updateCustomerBillingAddress);
-    });
-
-    it('fails when country is missing in customer payment details', async () => {
-      sandbox
-        .stub(stripeHelper, 'extractCustomerDefaultPaymentDetails')
-        .resolves({ postalCode: 'GD' });
-      sandbox.stub(stripeHelper, 'updateCustomerBillingAddress').resolves();
-      const actual = await stripeHelper.setCustomerLocation(customer1);
-      assert.isFalse(actual);
-      sinon.assert.calledOnceWithExactly(
-        stripeHelper.extractCustomerDefaultPaymentDetails,
-        customer1
-      );
-      sinon.assert.calledOnce(Sentry.withScope);
-      sinon.assert.calledOnceWithExactly(
-        sentryScope.setContext,
-        'setCustomerLocation',
-        { customer: { id: customer1.id } }
-      );
-      sinon.assert.calledOnceWithExactly(
-        Sentry.captureMessage,
-
-        `Cannot find a postal code or country for customer ${customer1.id}`,
-        Sentry.Severity.Error
-      );
-      sinon.assert.notCalled(stripeHelper.googleMapsService.getStateFromZip);
-      sinon.assert.notCalled(stripeHelper.updateCustomerBillingAddress);
-    });
-
     it('fails when an error is thrown by Google Maps service', async () => {
-      sandbox
-        .stub(stripeHelper, 'extractCustomerDefaultPaymentDetails')
-        .resolves({ postalCode: '99999', country: 'GD' });
       sandbox.stub(stripeHelper, 'updateCustomerBillingAddress').resolves();
       mockGoogleMapsService.getStateFromZip = sandbox.stub().rejects(err);
-      const actual = await stripeHelper.setCustomerLocation(customer1);
+      const actual = await stripeHelper.setCustomerLocation({
+        customerId: customer1.id,
+        postalCode: expectedAddressArg.postalCode,
+        country: expectedAddressArg.country,
+      });
       assert.isFalse(actual);
-      sinon.assert.calledOnceWithExactly(
-        stripeHelper.extractCustomerDefaultPaymentDetails,
-        customer1
-      );
```

Comment:
Can we assert that Sentry.withScope and scope.setContext are called with the expected values here?

Reply:
The argument passed to Sentry.withScope is a callback function that uses scope.setContext.  The test is already verifying the arguments for scope.setContext.  But since we don't have a reference to the callback function we cannot do an exact match there.

Reply:
https://github.com/mozilla/fxa/pull/11490/files#diff-b2ea299307ae708731a29f26e148068eab268e1c06e58723c636bdfcf956704eR4540 is where setContext's call is asserted.